### PR TITLE
Add server side code for posting comments

### PR DESCRIPTION
**Pull request by CarlosM10011**
This pull request adds the server side code for posting comments on the server.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -30,21 +28,26 @@
...
     String output = gson.toJson(this.comments);
-    response.setContentType(this.CONTENT_TYPE);
+    response.setContentType(this.GET_CONTENT_TYPE);
     response.getWriter().println(output);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = request.getParameter("name");
+    String body = request.getParameter("body");
```
Comment (tinagrasso) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Create constants for these strings.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -30,21 +28,26 @@
...
+    if (name == null || body == null) { // Someone sending a bad form.
```
Comment (tinagrasso) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: What about empty strings?

**Review by tinagrasso — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -42,12 +42,21 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String name = request.getParameter("name");
-    String body = request.getParameter("body");
-    if (name == null || body == null) { // Someone sending a bad form.
-      return;
+    final String name = request.getParameter("name");
+    final String body = request.getParameter("body");
+    if (!isSaneForm(name, body)) {
```
Comment (tinagrasso) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:49]: Consider adding an error response code and message. 4XX errors are for client errors.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -42,12 +42,21 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String name = request.getParameter("name");
-    String body = request.getParameter("body");
-    if (name == null || body == null) { // Someone sending a bad form.
-      return;
+    final String name = request.getParameter("name");
+    final String body = request.getParameter("body");
```
Comment (tinagrasso) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Constants for param fields.

**Review by tinagrasso — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -30,21 +28,38 @@
...
+      return;
+    }
+    this.comments.add(new Comment(name, new Date(), body));
+    response.sendRedirect(this.POST_REDIRECT_URL);
+  }
+
+  /** Checks to see if the comment form sent via the html post is valid. */
+  private Boolean isSaneForm(String name, String body) {
+    if ((name != null && !name.isEmpty())
+        && (body != null && !body.isEmpty())) { // Someone sending a bad form.
```
Comment (tinagrasso) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: You can simply return the if statement.

```
return (name != null && !name.isEmpty()) && (body != null && !body.isEmpty());
```

**Review by tinagrasso — CHANGES_REQUESTED** [review]

**Review by hzimz — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -30,21 +28,35 @@
...
+    response.setContentType(this.GET_CONTENT_TYPE);
     response.getWriter().println(output);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    final String name = request.getParameter(POST_NAME_PARAMETER);
+    final String body = request.getParameter(POST_MESSAGE_PARAMETER);
+    if (!isSaneForm(name, body)) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
```
Comment (tinagrasso) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Error message?

**Review by tinagrasso — CHANGES_REQUESTED** [review]

**Review by tinagrasso — APPROVED** [review]

**Pull request merged**